assembl/static2/js/app/actions/authenticationActions.js

```diff
@@ -0,0 +1,63 @@
+import { signUp, changePasswordRequest } from '../services/authenticationService';
+
+
+const signupSuccess = () => {
+  return {
+    type: 'SIGNUP_SUCCESS'
+  };
+};
+
+const signupIncorrectPassword = () => {
+  return {
+    type: 'SIGNUP_INCORRECT_PASSWORD'
+  };
+};
+
+const signupGeneralError = (error) => {
+  return {
+    type: 'SIGNUP_GENERAL_ERROR',
+    error: error
+  };
+};
+
+export const signupAction = (payload) => {
+  return (dispatch) => {
+    return signUp(payload).then(() => {
+      dispatch(signupSuccess());
+    }).catch((error) => {
+      // Should match on a more generic thing that accounts for JS errors +
+      // server returned errors
+      switch (error.name) {
```

Comment:
I think I understand what you mean. I have, for example, created an exception class for mismatched passwords. Do you mean we should seek to create more exception classes that show specific conditions and match on those classes?